src/main.cpp

```diff
@@ -19,16 +19,16 @@ bool wasAppStartedFromARemoteDrive()
     const std::string exePath = kiwix::getExecutablePath();
 
     // Usual mounting point for Windows Network Drives
-    if ( exePath.substr(0, 2) == "\\\\" )
+    if ( exePath.length() >= 2 && exePath.substr(0, 2) == "\\\\" )
         return true;
 
     // Support macOS Parallels shared folders.
     // See https://kb.parallels.com/130138
-    if ( exePath.substr(0, 7) == "C:\\Mac\\" )
+    if ( exePath.length() >= 7 && exePath.substr(0, 7) == "C:\\Mac\\" )
         return true;
 
     // Last chance to identify as remote drive
-    return GetDriveTypeA(exePath.substr(0, 3).c_str()) == DRIVE_REMOTE;
+    return exePath.length() >= 3 && GetDriveTypeA(exePath.substr(0, 3).c_str()) == DRIVE_REMOTE;
 }
```

Comment:
This entire commit is not needed. [`std::string::substr()`](https://en.cppreference.com/w/cpp/string/basic_string/substr) is safe on the overflow caused by its second parameter - it then simply returns the substring until the end of the string.

> Returns a substring `[pos, pos + count)`. If the requested substring extends past the end of the string, i.e. the count is greater than `size() - pos` (e.g. if `count == npos`), the returned substring is `[pos, size())`.

Reply:
@veloman-yunkan I will remove it and merge if OK for you?

Reply:
@kelson42 That's OK